torchao/quantization/GPTQ.py

```diff
@@ -762,11 +762,15 @@ def _check_linear_int4_k(k, groupsize = 1, inner_k_tiles = None):
     return k_divisible_by_groupsize
 
 def linear_forward_int4(x, weight_int4pack, scales_and_zeros, out_features, groupsize):
```

Comment:
So these conversions to bfloat16 are primarily needed because of `_weight_int4pack_mm`?

Reply:
yes, that's correct